src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -183,14 +188,17 @@ class CMutableDataset:
 
     def __new__(
         cls,
-        data: Union[Mapping[str, np.ndarray], Mapping[str, Union[np.ndarray, Mapping[str, np.ndarray]]]],
-        dataset_type: Optional[str] = None,
+        data: Union[
+            Mapping[ComponentType, np.ndarray],
+            Mapping[ComponentType, Union[np.ndarray, Mapping[str, np.ndarray]]],
+        ],
+        dataset_type: Any = None,
     ):
         instance = super().__new__(cls)
         instance._mutable_dataset = MutableDatasetPtr()
         instance._buffer_views = []
 
-        instance._dataset_type = dataset_type if isinstance(dataset_type, str) else get_dataset_type(data)
+        instance._dataset_type = dataset_type if dataset_type in list(DataType) else get_dataset_type(data)
```

Comment:
This would mean we are doing `list` every single time if we want to check something. Instead, define a method in `DataType`, something like:
```
    @classmethod
    def contains(cls, a):
        return a in cls._member_map_
```
so that we could do:
```suggestion
        instance._dataset_type = dataset_type if DataType.contains(dataset_type) else get_dataset_type(data)
```

Reply:
@Jerry-Jinfeng-Guo You are correct. Thanks for the suggestion.